.github/workflows/scripts/wait-for-merge-commit.js

```diff
@@ -0,0 +1,57 @@
+const MAX_WAIT_MS = 10 * 1000; // 10s
+const WAIT_PERIOD = 0.5 * 1000; // 0.5s
+
+function wait(interval) {
+  return new Promise(resolve => {
+    setTimeout(resolve, interval);
+  });
+}
+
+async function waitForMergeCommit({github, context, core}) {
+  const pullRequest = context.payload.pull_request;
+  const pullNumber = pullRequest.number;
+
+  let timedOut = false;
+  let mergeable = false;
+  let mergeCommitSha = null;
+
+  const start = new Date().getTime();
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const response = await github.rest.pulls.get({
+      ...context.repo,
+      pull_number: pullNumber,
+    });
+    if (response.status === 200) {
+      mergeable = response.data.mergeable;
```

Comment:
if mergeable is ever false you can stop immediately since it's a merge conflict

Reply:
Logic has changed to wait until mergable is non-null and then exit the loop.

I've changed the sentry sha to fallback to the head sha if there is no merge commit, which seems to happen on this PR, maybe because the PR cannot be merged until it's approved?

The logs for both the wait for merge script and the getsentry dispatch should help us quickly debug this in the future too if issues arise.

Reply:
OK fixed my broken logic that meant I wasn't getting the commit.

I've cause this script to fail if the PR isn't mergable so the mergeCommitSha should be a reliable output.